### PR TITLE
fix: AWS deployment issues

### PR DESCRIPTION
**Pull request by mevech**
Have tried out the AWS deployment and encountered a few issues related to dart version and ACL Not Supported issues.
It also address that RDS request ssl by default, else the service will not start

### Related issues
closes: #2622
closes: #2620
closes: #2619

## Pre-launch Checklist

- [x] I read the [Contribute](https://docs.serverpod.dev/contribute) page and followed the process outlined there for submitting PRs.
- [x] This update contains only one single feature or bug fix and nothing else. (If you are submitting multiple fixes, please make multiple PRs.)
- [x] I read and followed the [Dart Style Guide](https://dart.dev/guides/language/effective-dart/style) and formatted the code with [dart format](https://dart.dev/tools/dart-format).
- [x] I listed at least one issue that this PR fixes in the description above.
- [ ] I updated/added relevant documentation (doc comments with `///`), and made sure that the documentation follows the same style as other Serverpod documentation. I checked spelling and grammar.
- [ ] I added new tests to check the change I am making.
- [ ] All existing and new tests are passing.
- [x] Any breaking changes are documented below.

If you need help, consider asking for advice on the [discussion board](https://github.com/serverpod/serverpod/discussions).

## Breaking changes

N/A

### CODE REVIEW

**Comment by mevech** [comment]
I have read the CLA Document and I hereby sign the CLA

**Comment by mevech** [comment]
recheck

**Comment by mevech** [comment]
- - -
   ***I have read the CLA Document and I hereby sign the CLA***
   - - -

**Comment by mevech** [comment]
recheck

**Comment by mevech** [comment]
- - -
   ***I have read the CLA Document and I hereby sign the CLA***
   - - -

**Comment by mevech** [comment]
recheck

**Comment by Aygistov** [comment]
Thank you so much!!! This commit saved my life)

File: templates/serverpod_templates/projectname_server_upgrade/deploy/aws/terraform/code-deploy.tf [hunk 1]
```diff
@@ -89,3 +89,10 @@ resource "aws_s3_bucket_acl" "deployment" {
   bucket = aws_s3_bucket.deployment.id
   acl    = "private"
 }
+
+resource "aws_s3_bucket_ownership_controls" "deployment" {
+  bucket = aws_s3_bucket.deployment.id
+  rule {
+    object_ownership = "ObjectWriter"
+  }
+}
```
Comment (Isakdl) [inline templates/serverpod_templates/projectname_server_upgrade/deploy/aws/terraform/code-deploy.tf:98]: Question: Just for my understanding, what does this do exactly?

**Review by Isakdl — COMMENTED** [review]
Looking good, very nice fix! Will help a lot of people :) 

I have a single question, on top of that we should also document how to update the dart version in our AWS deploy docs!

File: templates/serverpod_templates/projectname_server_upgrade/deploy/aws/terraform/code-deploy.tf (hunk 1 above)
Comment (mevech) [reply templates/serverpod_templates/projectname_server_upgrade/deploy/aws/terraform/code-deploy.tf:98]: aws s3 bucket by default have ACL disabled. To enable ACL, we need to set the controlling ownership.

there are 2 types of ownership:
`BucketOwnerPreferred` allow the writter to transfer ownership to bucket owner, and `ObjectWritter` will own the object but can grant object access permissions to others.  :)

Comment (mevech) [reply templates/serverpod_templates/projectname_server_upgrade/deploy/aws/terraform/code-deploy.tf:98]: further reads:
when they introduced it: https://aws.amazon.com/about-aws/whats-new/2021/11/amazon-s3-object-ownership-simplify-access-management-data-s3/
terraform explanation: https://registry.terraform.io/providers/hashicorp/aws/latest/docs/resources/s3_bucket_ownership_controls
aws explanation:
https://docs.aws.amazon.com/AmazonS3/latest/userguide/about-object-ownership.html

**Pull request merged**